These failures are for input files 

```
[INFO]      [echo] Checkstyle started (checkstyle-non-main-files-checks.xml): 13/07/2026 06:22:09 PM
[INFO] [checkstyle] Running Checkstyle  on 5782 files
[ERROR] [checkstyle] [ERROR] /home/circleci/project/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/rulewrappinglines/InputWrappingLinesLineLimit.java:7: Line should not be longer than 100 symbols [lineLength]
[ERROR] [checkstyle] [ERROR] /home/circleci/project/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/rulewrappinglines/InputWrappingLinesLineLimit.java:13: Line should not be longer than 100 symbols [lineLength]
[ERROR] [checkstyle] [ERROR] /home/circleci/project/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/rulewrappinglines/InputWrappingLinesLineLimit.java:17: Line should not be longer than 100 symbols [lineLength]
```

<img width="1244" height="514" alt="image" src="https://github.com/user-attachments/assets/179cf796-0a35-46f7-9bf0-1c141e3be150" />

--- a/config/checkstyle-non-main-files-suppressions.xml
+++ b/config/checkstyle-non-main-files-suppressions.xml
@@ -76,6 +76,8 @@
   <suppress id="lineLength"
          files="src[\\/]it[\\/]resources[\\/].*[\\/]InputNoLineWrapping\.java"/>
   <suppress id="lineLength"
+         files="src[\\/]it[\\/]resources[\\/].*[\\/]InputWrappingLinesLineLimit\.java"/>
+  <suppress id="lineLength"
          files="src[\\/]it[\\/]resources[\\/].*[\\/]InputFormattedNoLineWrapping\.java"/>
   <suppress id="lineLength"
          files="src[\\/]test[\\/]resources[\\/].*[\\/]InputLineLength.*\.java"/>

--- a/src/it/java/com/openjdk/checkstyle/test/chapterformatting/rulewrappinglines/WrappingLinesTest.java
+++ b/src/it/java/com/openjdk/checkstyle/test/chapterformatting/rulewrappinglines/WrappingLinesTest.java
@@ -23,7 +23,7 @@ import org.junit.jupiter.api.Test;
 
 import com.openjdk.checkstyle.test.base.AbstractOpenJdkModuleTestSupport;
 
-public class OneStatementPerLineTest extends AbstractOpenJdkModuleTestSupport {
+public class WrappingLinesTest extends AbstractOpenJdkModuleTestSupport {
 
     @Override
     public String getPackageLocation() {
@@ -48,6 +48,21 @@ public class OneStatementPerLineTest extends AbstractOpenJdkModuleTestSupport {
     @Test
     public void testOneStatementPerLineForLoop() throws Exception {
         verifyWithWholeConfig(getPath("InputOneStatementPerLineForLoop.java"));
+    }
+
+    @Test
+    public void testWrappingLinesLineLimit() throws Exception {
+        verifyWithWholeConfig(getPath("InputWrappingLinesLineLimit.java"));
+    }
+
+    @Test
+    public void testWrappingLinesDosAndDontsOne() throws Exception {
+        verifyWithWholeConfig(getPath("InputWrappingLinesDosAndDontsOne.java"));
+    }
+
+    @Test
+    public void testWrappingLinesDosAndDontsTwo() throws Exception {
+        verifyWithWholeConfig(getPath("InputWrappingLinesDosAndDontsTwo.java"));
     }
 
 }

--- a/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/ruleimportstatements/importorder/InputImportOrderInsideStaticGroups.java
+++ b/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/ruleimportstatements/importorder/InputImportOrderInsideStaticGroups.java
@@ -6,12 +6,12 @@ import static java.lang.Math.PI;
 import static java.lang.Math.abs;
 import static org.mockito.Mockito.mock;
 import static java.util.Collections.emptyList;
-// violation above 'Import statement for 'java.util.Collections.emptyList'
-// violates the configured import group order.'
+// violation above """Import statement for 'java.util.Collections.emptyList'
+// violates the configured import group order."""
 
 import javax.swing.JButton;
-// violation above 'Import 'javax.swing.JButton' violates the configured
-// relative order between static and non-static import'
+// violation above """Import 'javax.swing.JButton' violates the configured
+// relative order between static and non-static import"""
 
 /**
  * Test file for static import order.

--- a/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/ruleimportstatements/importorder/InputImportOrderStaticFirst.java
+++ b/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/ruleimportstatements/importorder/InputImportOrderStaticFirst.java
@@ -4,7 +4,8 @@ import static java.lang.Math.PI;
 import static java.lang.Math.max;
 
 import java.util.Arrays;
-// violation above 'Import 'java.util.Arrays' violates the configured relative order between static and non-static imports.'
+// violation above """Import 'java.util.Arrays' violates the configured relative order
+// between static and non-static imports."""
 
 /**
  * Invalid: static imports appear before non-static imports.

--- a/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/ruleimportstatements/importorder/InputImportOrderWrongPackageOrder.java
+++ b/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/ruleimportstatements/importorder/InputImportOrderWrongPackageOrder.java
@@ -4,7 +4,8 @@ import javax.swing.JButton;
 import javax.swing.JFrame;
 
 import java.util.Arrays;
-// violation above 'Import statement for 'java.util.Arrays' violates the configured import group order.'
+// violation above 'Import statement for 'java.util.Arrays' violates
+// the configured import group order.'
 
 /**
  * Invalid: javax.* imports appear before java.* imports.

--- a/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/rulewrappinglines/InputWrappingLinesDosAndDontsOne.java
+++ b/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/rulewrappinglines/InputWrappingLinesDosAndDontsOne.java
@@ -1,0 +1,110 @@
+package com.openjdk.checkstyle.test.chapterformatting.rulewrappinglines;
+
+import java.util.stream.Stream;
+import static java.util.stream.Collectors.joining;
+
+public class InputWrappingLinesDosAndDontsOne {
+
+    class Errors {
+        static Error invalidRepeatableAnnotationNotApplicable(boolean targetContainerType,
+                boolean on) {
+            return new Error("Error");
+        }
+
+        static Error invalidRepeatableAnnotationNotApplicableInContext(
+                boolean targetContainerType) {
+            return new Error("Error");
+        }
+
+    }
+
+    class Argument {
+        static String prettyPrint(String str) {
+            return "";
+        }
+    }
+
+    public void styleGuideDosLineLimit(boolean isTypeParam, String args) {
+        boolean targetContainerType = false;
+        boolean on = false;
+        Error e = isTypeParam
+                ? Errors.invalidRepeatableAnnotationNotApplicable(targetContainerType, on)
+                : Errors.invalidRepeatableAnnotationNotApplicableInContext(targetContainerType);
+
+        String pretty = Stream.of(args)
+                .map(Argument::prettyPrint)
+                .collect(joining(", "));
+    }
+
+    public void styleGuideDontsLineLimit(boolean isTypeParam, String args) {
+        boolean targetContainerType = false;
+        boolean on = false;
+
+        // checkstyle is a static tool it can not determine whether line wrapping increase
+        // readability or not
+        Error e = isTypeParam
+                ? Errors.invalidRepeatableAnnotationNotApplicable(
+                        targetContainerType, on)
+                : Errors.invalidRepeatableAnnotationNotApplicableInContext(
+                        targetContainerType);
+
+        // Not covered until https://github.com/checkstyle/checkstyle/issues/20714
+        String pretty = Stream.of(args).map(Argument::prettyPrint).collect(joining(", "));
+    }
+
+    public String withMany(int arg, int that, int needs) {
+        return "";
+    }
+
+    public void methodCall(String first, int second) {
+    }
+
+    public int to(int first, int second) {
+        return 0;
+    }
+
+    public void styleGuideDosSyntax(int arguments, int that, int needs, int be, int wrapped,
+            int tos, int avoid, int veryLong, int lines) {
+
+        methodCall(withMany(arguments, that, needs),
+                to(be, (wrapped - tos) * avoid / veryLong - lines));
+    }
+
+    public void styleGuideDontsSyntax(int arguments, int that, int needs, int be, int wrapped,
+            int tos, int avoid, int veryLong, int lines) {
+
+        // Not covered until https://github.com/checkstyle/checkstyle/issues/20716
+        methodCall(withMany(arguments, that, needs), to(be, (wrapped
+                - tos) * avoid / veryLong - lines));
+    }
+
+    public String styleGuideDosStatement() {
+        int i = 1;
+        int j = 2;
+        int k = 3;
+        boolean condition = false;
+        String expression = "";
+
+        i += j;
+        j += k;
+        if (condition) {
+            return expression;
+        }
+        return "";
+    }
+
+    public String styleGuideDontsStatement() {
+        int i = 1;
+        int j = 2;
+        int k = 3;
+        boolean condition = false;
+        String expression = "";
+
+        i += j; j += k; // violation 'Only one statement'
+
+        // violation below ''{' at column 24 should have line break after.'
+        if (condition) { return expression; }
+        return "";
+    }
+
+}

--- a/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/rulewrappinglines/InputWrappingLinesDosAndDontsTwo.java
+++ b/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/rulewrappinglines/InputWrappingLinesDosAndDontsTwo.java
@@ -1,0 +1,52 @@
+package com.openjdk.checkstyle.test.chapterformatting.rulewrappinglines;
+
+import java.util.stream.IntStream;
+
+public class InputWrappingLinesDosAndDontsTwo {
+
+    public void styleGuideDosLineContinue(int that, int takes, int a, int lengthy, int is,
+            int computed, int using, int thatTakes, int aLongList, int ofArguments, int complex,
+            int exp, int list, int of, int arguments) {
+
+        int[] numbers = new int[10];
+        int anInteger = amethod(that, takes,
+                a, lengthy, list, of, arguments);
+
+        // Variant 2
+        int anInteger1 = that * (is + computed) / using
+                                + a * complex - exp;
+
+        // Variant 3
+        int anInteger2 = amethod(thatTakes,
+                                aLongList,
+                                ofArguments);
+
+        // Variant 4
+        double anInteger3 = IntStream.of(numbers)
+                                     .mapToDouble(Math::sqrt)
+                                     .sum();
+    }
+
+    public void styleGuideDontsLineContinue(int that, int takes, int a, int lengthy,
+            int list, int of, int arguments) {
+
+        // Not covered until https://github.com/checkstyle/checkstyle/issues/20714
+        int anInteger = amethod(that,
+                                takes,
+                                a, lengthy, list,
+                                of, arguments);
+
+        if (somePredicate() || // violation ''\|\|' should be on a new line.'
+            someOtherPredicate()) {
+            System.out.println("Avoid");
+        }
+        // violation 3 lines above '.* incorrect indentation level 12, expected .* 16.'
+    }
+
+    private int amethod(int... args) { return 0; }
+
+    private boolean somePredicate() { return false; }
+
+    private boolean someOtherPredicate() { return false; }
+
+}

--- a/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/rulewrappinglines/InputWrappingLinesLineLimit.java
+++ b/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/rulewrappinglines/InputWrappingLinesLineLimit.java
@@ -1,0 +1,25 @@
+package com.openjdk.checkstyle.test.chapterformatting.rulewrappinglines;
+
+// violation 4 lines below 'Line is longer than 100 characters (found 103).'
+/**
+ * This is a short Javadoc comment.
+ * This is a very long url https://verylongurlexamplebutitdoesnotthrowanyerrorbecausetheURLisignoredbylineLengthCheck.com
+ * ThisJavadocCommentIsAReallyLongWordThatExceedsDefaultLineLimitOfHundredCharactersSoitwillgiveserror.
+ */
+public class InputWrappingLinesLineLimit {
+    void testMethod(String str) {
+        System.out.println("This is a short line.");
+
+        System.out.println("This line is long and exceeds the default limit of 100 characters............");
+        // violation above 'Line is longer than 100 characters (found 108).'
+
+        String str1 = """
+                    This is a very really long string that exceeds the limit it will gives violation..........................
+                    """;
+        // violation 2 lines above 'Line is longer than 100 characters (found 126).'
+
+        String str2 = """
+                    This is a very really long string that is exactly equal 100 so no violation....
+                    """;
+    }
+}

--- a/src/main/resources/openjdk_checks.xml
+++ b/src/main/resources/openjdk_checks.xml
@@ -52,6 +52,13 @@
     <property name="eachLine" value="true"/>
   </module>
 
+  <module name="LineLength">
+    <property name="fileExtensions" value="java"/>
+    <property name="max" value="100"/>
+    <property name="ignorePattern"
+          value="^package.*|^import.*|href\s*=\s*&quot;[^&quot;]*&quot;|http://|https://|ftp://"/>
+  </module>
+
   <module name="TreeWalker">
     <!-- Add TreeWalker checks based on OpenJDK style guide -->
     <module name="Indentation">

--- a/src/site/xdoc/checks/sizes/linelength.xml
+++ b/src/site/xdoc/checks/sizes/linelength.xml
@@ -349,6 +349,10 @@ class Example6 {
             Google Style</a>
           </li>
           <li>
+            <a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fopenjdk_checks.xml+repo%3Acheckstyle%2Fcheckstyle+LineLength">
+            OpenJDK Style</a>
+          </li>
+          <li>
             <a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fsun_checks.xml+repo%3Acheckstyle%2Fcheckstyle+LineLength">
             Sun Style</a>
           </li>

--- a/src/site/xdoc/checks/sizes/linelength.xml.template
+++ b/src/site/xdoc/checks/sizes/linelength.xml.template
@@ -147,6 +147,10 @@
             Google Style</a>
           </li>
           <li>
+            <a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fopenjdk_checks.xml+repo%3Acheckstyle%2Fcheckstyle+LineLength">
+            OpenJDK Style</a>
+          </li>
+          <li>
             <a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fsun_checks.xml+repo%3Acheckstyle%2Fcheckstyle+LineLength">
             Sun Style</a>
           </li>

--- a/src/site/xdoc/openjdk_style.xml
+++ b/src/site/xdoc/openjdk_style.xml
@@ -637,10 +637,48 @@
                   <span class="wrapper inline">
                     <img src="images/ok_blue.png" alt=""/>
                   </span>
+                  <a href="checks/sizes/linelength.html#LineLength">
+                    LineLength</a>
+                  (<a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fopenjdk_checks.xml+repo%3Acheckstyle%2Fcheckstyle+LineLength">
+                  config</a>)
+                  <br />
+                  Checkstyle is a static tool it can only apply strict column limit it can not
+                  apply column limit on basis of readability.
+                  <br />
+                  <br />
+                  <span class="wrapper inline">
+                    <img src="images/ban_red.png" alt=""/>
+                  </span>
+                  Wrapping at a higher syntactical level rule not covered until:
+                  <a href="https://github.com/checkstyle/checkstyle/issues/20716">
+                    #20716
+                  </a>
+                  <br />
+                  <br />
+                  <span class="wrapper inline">
+                    <img src="images/ok_blue.png" alt=""/>
+                  </span>
                   <a href="checks/coding/onestatementperline.html#OneStatementPerLine">
                     OneStatementPerLine</a>
                   (<a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fopenjdk_checks.xml+repo%3Acheckstyle%2Fcheckstyle+OneStatementPerLine">
                   config</a>)
+                  <br />
+                  <br />
+                  <span class="wrapper inline">
+                    <img src="images/ok_blue.png" alt=""/>
+                  </span>
+                  <a href="checks/misc/indentation.html#Indentation">
+                    Indentation</a>
+                  (<a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fopenjdk_checks.xml+repo%3Acheckstyle%2Fcheckstyle+Indentation">
+                  config</a>)
+                  <br />
+                  <span class="wrapper inline">
+                    <img src="images/ban_red.png" alt=""/>
+                  </span>
+                  Not completely covered until:
+                  <a href="https://github.com/checkstyle/checkstyle/issues/20714">
+                    #20714
+                  </a>
                 </td>
                 <td>
                   <a href="https://github.com/checkstyle/checkstyle/tree/master/src/it/resources/com/openjdk/checkstyle/test/chapterformatting/rulewrappinglines">

